modules/drivers/databricks/resources/metabase-plugin.yaml

```diff
@@ -12,27 +12,58 @@ driver:
     connection-properties:
       - merge:
         - host
-        - placeholder: 'xxxxxxxxxx.cloud.databricks.com'
+        - display-name: Host
+          required: true
+          placeholder: xxxxxxxxxx.cloud.databricks.com
+          helper-text: The Databricks host URL. Could also be adb-xxxxx.azuredatabricks.net
       - name: http-path
-        display-name: HTTP path
+        display-name: HTTP Path
         helper-text: Found in SQL Warehouses > Connection details
         required: true
+        placeholder: /sql/1.0/endpoints/abcdef1234567890
+      - name: use-m2m
+        display-name: Use Machine to Machine (M2M) authentication
+        type: boolean
+        default: false
```

Comment:
I believe making m2m new default is right decision. Wouldn't changing this  default put us in that position?

Reply:
Please help me find the error in my reasoning then:
- Setting the default to true here (line 27) will make m2m auth default in a sense that screen for adding database connection will provide this option as a default. User would have to explicitly opt out for PAT auth.
- _Database details_ of connections created before this change will be transformed into jdbc specs using PAT code path because `use-m2m` is `nil` in those cases.
- New _database connections_ will have `use-m2m` set either false or true, hence the right spec would be generated.
- Old database connections will get the `use-m2m` populated on successful edit.

Reply:
In your reasoning you are assuming that existing configurations will leave use-m2m as nil/null/none with my proposed changes. If this is true, then your reasoning holds. I was operating under the assumption that it would be set to the default value. I will change the `use-m2m` var default to `true`. Please resolve this if this makes sense.